index.ts

```diff
@@ -18,8 +18,48 @@ export class Tree<TValue extends number> {
   }
 
   private _insert(node: ITreeNode<TValue>, value: TValue) {
-    const direction = value >= node.value ? 'right' : 'left'
-    if (node[direction] !== undefined) this._insert(node[direction]!, value) // eslint-disable-line @typescript-eslint/no-non-null-assertion
+    const direction = this._setDirection(node, value)
+    /* eslint-disable */
+    if (node[direction] !== undefined) this._insert(node[direction]!, value)
     else node[direction] = { value }
+    /* eslint-enable */
+  }
+
+  private _setDirection(node: ITreeNode<TValue>, value: TValue) {
+    return value >= node.value ? 'right' : 'left'
+  }
+
+  public remove(value: TValue) {
+    if (this.root === undefined) throw new Error('There is no tree currently')
```

Comment:
```suggestion
    if (!this.root) throw new Error('There is no tree currently')
```
but actaully why you trow you can instead return boolean if the element was removed.

Reply:
throw error instead of boolean to inform that there is no tree as of now
need to insert tree before proceed to remove